### PR TITLE
Special-case "kxmlgui" package: don't make it auto-require buildreq-kde

### LINKED ISSUE #487: buildreq_ban not applying for kxmlgui
```
$ cat Makefile
PKG_NAME := kxmlgui
URL = https://download.kde.org/stable/frameworks/5.62/kxmlgui-5.62.0.tar.xz
ARCHIVES = 

include ../common/Makefile.common
$ cat buildreq_ban
# This file contains build requirements that get picked up but are
# undesirable. One entry per line, no whitespace.
buildreq-kde
```

But Autospec always adds the dependency, causing the build to fail since that makes kxmlgui depend on kxmlgui. Added from:

```python
    if "kde.org" in url or "https://github.com/KDE" in url:                                                                                     
        buildreq.add_buildreq("buildreq-kde")

### PR DESCRIPTION
**Pull request by thiagomacieira**
Fixes #487, by way of a hacky workaround.

### CODE REVIEW

File: autospec/tarball.py [hunk 1]
```diff
@@ -471,7 +471,8 @@ def name_and_version(name_arg, version_arg, filemanager):
         buildreq.add_buildreq("buildreq-gnome")
 
     if "kde.org" in url or "https://github.com/KDE" in url:
-        buildreq.add_buildreq("buildreq-kde")
+        if "kxmlgui" not in url:
+            buildreq.add_buildreq("buildreq-kde")
```
Comment (bryteise) [inline autospec/tarball.py]: I assume there is some ordering issue preventing kxmlgui from buildreq_banning buildreq-kde?

Comment (thiagomacieira) [reply autospec/tarball.py]: Probably, I don't know why buildreq_ban does not apply.

Comment (phmccarty) [reply autospec/tarball.py]: It is an ordering issue... a PR will be opened shortly.

Comment (bryteise) [reply autospec/tarball.py]: With @phmccarty change merged I'm closing this as buildreq_ban will apply to urls now too.

**Comment by thiagomacieira** [comment]
Thanks.

**Pull request closed without merging**